presto-main/src/main/java/com/facebook/presto/sql/planner/plan/TableScanNode.java

```diff
@@ -74,23 +76,43 @@ public TableScanNode(
             List<Symbol> outputs,
             Map<Symbol, ColumnHandle> assignments,
             Optional<TableLayoutHandle> tableLayout,
-            @Nullable TupleDomain<ColumnHandle> currentConstraint)
+            TupleDomain<ColumnHandle> currentConstraint,
+            TupleDomain<ColumnHandle> enforcedConstraint)
+    {
+        this(id, table, outputs, assignments, tableLayout, currentConstraint, enforcedConstraint, false);
+    }
+
+    private TableScanNode(
+            PlanNodeId id,
+            TableHandle table,
+            List<Symbol> outputs,
+            Map<Symbol, ColumnHandle> assignments,
+            Optional<TableLayoutHandle> tableLayout,
+            TupleDomain<ColumnHandle> currentConstraint,
+            TupleDomain<ColumnHandle> enforcedConstraint,
+            boolean internal)
```

Comment:
This flag here is a code smell. Looking at the rest of the commit, it seems to be used only in the static factory method above. Move the check there, instead.